### PR TITLE
feat: return error if guesses aren't in wordlist or unique

### PR DESCRIPTION
**Pull request by pkej**
Only allow submission if suggested word is in the wordlist, no more fake words allowed, return "Invalid guess". Only allow new words to guesses, if current guess is the same as an earlier word, return "Invalid guess"

### CODE REVIEW

File: server/api/guess.ts [hunk 1]
```diff
@@ -4,19 +4,35 @@ import MemoryDriver from 'unstorage/drivers/memory'
 
 import wordList from 'wordlist-english/index.js'
 const validWords = wordList['english/10'].filter(word => word.length === 5)
-
+//console.log(validWords)
```
Comment (danielroe) [inline server/api/guess.ts:7]:
```suggestion

```

File: server/api/guess.ts [hunk 2]
```diff
@@ -4,19 +4,35 @@ import MemoryDriver from 'unstorage/drivers/memory'
...
+  if (!validWords.includes(guess)) {
     return createError({
       statusCode: 422,
-      message: 'Invalid guess',
+      message: 'Invalid guess, word not in word list',
+    })
+  }
+const stateWords = [].concat.apply([], state);
+
+if(stateWords.includes(guess))  {
```
Comment (danielroe) [inline server/api/guess.ts:32]:
```suggestion
if(state.some(([word]) => word === guess))  { 
```

**Review by danielroe — COMMENTED** [review]
Thank you - this looks great! Two small suggestions...

**Comment by pkej** [comment]
Great suggestions, I'll see what I can learn from the second one

**Comment by pkej** [comment]
Could you explain line 29 in guess.ts? I can't quite get how it works.

**Review by danielroe — APPROVED** [review]
Great improvement - thank you!

**Comment by danielroe** [comment]
> Could you explain line 29 in guess.ts? I don't understand how it works, but truthfully, my own version was just cut and paste code, and I don't think I understand that either.

[`.some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) is a semantic way of returning a boolean if a condition is true, and it means we don't have to create a separate array to iterate over. The game state is an array of [word, hint] pairs, which is why we destructure the word out of the state.

**Pull request merged**